app/registries/pagelets/manager.rb

```diff
@@ -55,6 +61,42 @@ def add_pagelet(*args)
       def pagelets_at(*args)
         @manager.pagelets_at(*args.unshift(@key))
       end
+
+      def with_profile(id, label, opts, &block)
+        profile = Profile.new(id, label, self, opts)
+        profile.instance_exec(&block)
+        profile.pagelets.each do |pagelet|
+          add_pagelet(*pagelet)
+        end
+      end
+    end
+
+    class Profile
+      attr_reader :id, :label, :opts, :pagelets
+
+      def initialize(id, label, context, opts)
+        @id = id
+        @label = label
+        @context = context
+        @opts = opts
+        @pagelets = []
+      end
+
+      def default?
+        @opts[:default]
+      end
+
+      def add_pagelet(mountpoint, opts)
+        opts = opts.merge(profiles: [self])
+        @pagelets << [mountpoint, opts]
+      end
+
+      def use_pagelet(mountpoint, opts)
```

Comment:
I don't think we need to pass an open `opts` object. We are using only `:key` from it.
```suggestion
      def use_pagelet(mountpoint, key)
```